packages/chain-adapters/src/utxo/UTXOBaseAdapter.ts

```diff
@@ -0,0 +1,159 @@
+import { CAIP2, caip2, caip19 } from '@shapeshiftoss/caip'
+import { bip32ToAddressNList, HDWallet, PublicKey } from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, chainAdapters, ChainTypes, UtxoAccountType } from '@shapeshiftoss/types'
+import { bitcoin } from '@shapeshiftoss/unchained-client'
+import WAValidator from 'multicoin-address-validator'
+
+import { ChainAdapter as IChainAdapter } from '../api'
+import { ErrorHandler } from '../error/ErrorHandler'
+import {
+  accountTypeToScriptType,
+  bnOrZero,
+  convertXpubVersion,
+  toRootDerivationPath
+} from '../utils'
+
+export type UTXOChainTypes = ChainTypes.Bitcoin // to be extended in the future to include other UTXOs
+
+export interface ChainAdapterArgs {
+  providers: {
+    http: bitcoin.api.V1Api //?
```

Comment:
I am assuming we want to make this more generic, but from what I can tell there is not an interface that would allow this for a generic UTXO unchained provider.  In the absence of that, should we leave this as-is for now or move it up to the implementing `BitcoinChainAdapter `class

Reply:
i think this is ok for now - but will let @kaladinlight chime in here, as these types come from unchained

Reply:
This is matching how cosmos base adapter is currently, but you are correct that each coinstack will have their own api client that would need to be controlled by the implementing class. As we only support btc right now, we can start here and continue to iterate

Reply:
okay, going to resolve this conversation and leave as is for now.  I will add a comment to the codebase.